pandas/core/indexes/base.py

```diff
@@ -5499,6 +5526,9 @@ def map(self, mapper, na_action=None):
             # empty
             attributes["dtype"] = self.dtype
 
+        if self._is_numeric_index() and is_numeric_dtype(new_values.dtype):
```

Comment:
what is the purpose here? (e.g. why is this not simply handled in the constructor)

Reply:
If both are NumericIndex, we'd want a NumericIndex we'd want a NumericInde with the appropriate dtype, else we'd want a Int64Index etc.

I'm not very fond of this either, but seems to me unavoidable, as long as we both support `NumericIndex` and Int|Float|Uint64Index.

Reply:
I've added this as a class attribute instead. That's a bit simpler.

Reply:
Is this ok?